stdlib/public/core/StringGuts.swift

```diff
@@ -910,6 +910,18 @@ extension _StringGuts {
     _invariantCheck()
   }
 
+  @inlinable internal
+  init(_initialCapacity capacity: Int) {
+    if capacity <= _SmallUTF8String.capacity {
+      self.init()
+    } else {
+      let storage = _SwiftStringStorage<UInt8>.create(
+        capacity: capacity,
+        count: 0)
+      self.init(_large: storage)
+    }
+  }
+
```

Comment:
I want to inline `if capacity <= _SmallUTF8String.capacity` because, if we only use this in string interpolation, `capacity` will always be a constant and the optimizer can fold away one of the branches. I also want to inline `self.init()` because it should reduce down to the empty string singleton constant.

I don't really care about what's in the `else` branch, though (although ideally it would skip any small-string tests). If this implementation inlines more than you want to, you can choose a different one.